bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -606,15 +606,19 @@ void PreProcessor::messageHandler<PreProcessReplyMsg>(MessageBase *msg) {
 
 void PreProcessor::msgProcessingLoop() {
   while (!msgLoopDone_) {
-    int count = 0;
     {
       std::unique_lock<std::mutex> l(msgLock_);
-      while (!msgLoopDone_ && msgs_.empty()) msgLoopSignal_.wait(l);
+      while (!msgLoopDone_ && msgs_.empty()) {
```

Comment:
Don't you like to replace msgs_.empty() by !msgs_. read_available()

Reply:
Yes, its on the way